slashing-protection/referencetests/src/test/java/tech/pegasys/web3signer/slashingprotection/ReferenceTestRunner.java

```diff
@@ -0,0 +1,145 @@
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Resources;
+import com.opentable.db.postgres.embedded.EmbeddedPostgres;
+import dsl.SignedArtifacts;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.flywaydb.core.Flyway;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.jupiter.api.Test;
+
+@SuppressWarnings("UnusedVariable")
+public class ReferenceTestRunner {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final ObjectMapper objectMapper =
+      new ObjectMapper()
+          .registerModule(new InterchangeModule())
+          .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS);
+  private final ValidatorsDao validators = new ValidatorsDao();
+
+  private EmbeddedPostgres setup() throws IOException {
+    final EmbeddedPostgres slashingDatabase = EmbeddedPostgres.start();
+
+    final Flyway flyway =
+        Flyway.configure()
+            .locations("/migrations/postgresql/")
+            .dataSource(slashingDatabase.getPostgresDatabase())
+            .load();
+    flyway.migrate();
+
+    return slashingDatabase;
+  }
+
+  @Test
+  void execute() throws IOException {
+    final URL refTestPath =
+        Resources.getResource(
+            Path.of("slashing-protection-interchange-tests", "tests", "generated").toString());
+
+    final Path testFilesPath = Path.of(refTestPath.getPath());
+    try (Stream<Path> testFiles = Files.list(testFilesPath)) {
+      testFiles.forEach(this::executeFile);
+    }
+  }
+
+  private void executeFile(final Path inputFile) {
+    LOG.info(inputFile.toString());
+    try {
+      final TestFileModel model = objectMapper.readValue(inputFile.toFile(), TestFileModel.class);
+      final String interchangeContent =
+          objectMapper.writeValueAsString(model.getInterchangeContent());
+
+      final EmbeddedPostgres db = setup();
+      final String databaseUrl =
+          String.format("jdbc:postgresql://localhost:%d/postgres", db.getPort());
+
+      final SlashingProtection slashingProtection =
+          SlashingProtectionFactory.createSlashingProtection(databaseUrl, "postgres", "postgres");
+
+      slashingProtection.importData(new ByteArrayInputStream(interchangeContent.getBytes(UTF_8)));
+
+      final List<String> validatorsInImport =
+          model.getInterchangeContent().getSignedArtifacts().stream()
+              .map(SignedArtifacts::getPublicKey)
+              .collect(Collectors.toList());
+
+      final Jdbi jdbi = DbConnection.createConnection(databaseUrl, "postgres", "postgres");
+      final List<String> publicKeysInDb =
+          jdbi.inTransaction(
```

Comment:
does this need to happen in a transaction?

Reply:
how else do you do it?